src/merkle_tree/mod.rs

```diff
@@ -133,12 +135,121 @@ impl Pushable for &MerkleTreeTwinProof {
     }
 }
 
+impl MerkleTreeTwinProof {
+    /// Convert a stwo Merkle proof into twin proofs for each pairs of queries.
+    pub fn from_stwo_proof(
+        logn: usize,
+        queries_parents: &[usize],
+        values: &[Vec<BaseField>],
+        merkle_decommitment: &MerkleDecommitment<BWSSha256MerkleHasher>,
+    ) -> Vec<Self> {
+        // find out all the queried positions and sort them
+        let mut queries = vec![];
+        for &queries_parent in queries_parents.iter() {
+            queries.push(queries_parent << 1);
+            queries.push((queries_parent << 1) + 1);
+        }
+        queries.sort_unstable();
+        queries.dedup();
+
+        // get the number of columns
+        let column_num = values.len();
+
+        // create the new value map
+        let mut queries_values_map = HashMap::new();
+        for (idx, &query) in queries.iter().enumerate() {
+            let mut v = vec![];
+            for value in values.iter().take(column_num) {
+                v.push(value[idx]);
+            }
+            queries_values_map.insert(query, v);
+        }
+
+        // require the column witness to be empty
+        assert!(merkle_decommitment.column_witness.is_empty());
```

Comment:
I see that multiple operations can fail in this function, if the assertions are not matched.
Hence I am wondering if the function should be more a `try_from` than a `from` and return a `Result`.